File: IM.Production.Domain/Bank/BaseBank.cs
﻿using Epam.ImitationGames.Production.Domain.Base;

namespace Epam.ImitationGames.Production.Domain.Bank
{
    /// <summary>
    /// Базовый класс для сущностей, связанных с банковскими операциями.
    /// </summary>
    public abstract class BaseBank : BaseChanging, IVisibleEntity
    {
        public abstract string DisplayName { get; }

        protected BaseBank(GameTime time, Customer customer, string description = null)
            : base(time, customer, description)
        {
        }
    }
}

File: src/IM.Production.UI/src/models/Bank/BaseBank.ts
import { BaseChanging } from "../Base/BaseChanging";
import { IVisibleEntity } from "../Base/IVisibleEntity";
import { GameTime } from "../Base/GameTime";
import { Customer } from "../Customer/Customer";

export abstract class BaseBank extends BaseChanging implements IVisibleEntity {
    constructor(
        public displayName: string,
        public customer: Customer,
        public description: string,
        public time: GameTime,
        public id: any
    )
    {
        super(time, customer, description, id);
    }
}
